web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -51,40 +48,104 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [nameAvailability]);
+  const emptyNameAvailability: NameAvailability = {
+    similarNames: [],
+    status: undefined,
+    lastUpdatedTimeStamp: "",
+  };
 
   const Unavailable = props.unavailable;
   const Available = props.available;
-  const onSubmit = props.onSubmit;
 
   const doSearch = useCallback(
     async (
       event: FormEvent<HTMLFormElement> | undefined,
       { isInitialSubmit }: { isInitialSubmit: boolean }
     ): Promise<void> => {
       searchBusinessName(event)
-        .then(({ nameAvailability, submittedName }) => {
-          setNameAvailability(nameAvailability);
-          setSubmittedName(submittedName);
-          if (onSubmit) {
-            onSubmit(submittedName, nameAvailability, isInitialSubmit);
+        .then(async ({ nameAvailability, submittedName }) => {
+          if (props.isDba) {
+            if (!nameAvailability || !updateQueue) {
+              return;
+            }
+            setNameAvailability(nameAvailability);
+            if (nameAvailability.status === "AVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                  },
+                  dbaBusinessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  nexusDbaName: submittedName,
+                })
+                .update();
+            }
+          } else {
+            if (!nameAvailability || !updateQueue || isInitialSubmit) {
+              return;
+            }
+            setFieldsInteracted([FIELD_NAME]);
+            if (nameAvailability.status === "AVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                    businessName: submittedName,
+                  },
+                  businessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  businessName: submittedName,
+                  nexusDbaName: emptyProfileData.nexusDbaName,
+                  needsNexusDbaName: emptyProfileData.needsNexusDbaName,
+                })
+                .update();
+            } else if (nameAvailability.status === "UNAVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                    businessName: submittedName,
+                  },
+                  businessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  businessName: submittedName,
+                  needsNexusDbaName: true,
+                })
```

Comment:
90% of these updates are in common between the AVAILABLE and UNAVAILABLE case. It seems like this duplication could be refactored by queuing the shared changes first and then using the if/else structure only to change the things are are actually different between the two cases

Reply:
simplified